azurerm/resource_arm_application_insights_analytics_item.go

```diff
@@ -0,0 +1,219 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+
+	"github.com/Azure/azure-sdk-for-go/services/appinsights/mgmt/2015-05-01/insights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmApplicationInsightsAnalyticsItem() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApplicationInsightsAnalyticsItemCreate,
+		Read:   resourceArmApplicationInsightsAnalyticsItemRead,
+		Update: resourceArmApplicationInsightsAnalyticsItemUpdate,
+		Delete: resourceArmApplicationInsightsAnalyticsItemDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
could we add some validation for this:

```suggestion
				ForceNew: true,
				ValidateFunc: validation.NoEmptyStrings,
```